### DISCUSSION TITLE
Login with amazon

### DISCUSSION THREAD

**Discussion by My-CD-ROM** [saleweaver/python-amazon-sp-api#313] [Q&A] [answered]
Hello. I am trying to get access for my clients.
When I make Login With Amazon request i received a "selling_partner_id" and "spapi_oauth_code".
Next request is goes for refresh token thet we are going to use in future request
authorize_auth_code(spapi_oauth_code)
In next requests I need to pu credentials

credentials = dict(
refresh_token="Some token",
lwa_app_id=os.environ.get('LWA_CLIENT_ID'),
lwa_client_secret=os.environ.get('LWA_CLIENT_SECRET'),
aws_access_key="",
aws_secret_key="",
role_arn=os.environ.get('SP_API_ROLE_ARN')
)

Where I can get "aws_access_key" and "aws_secret_key" from client if amazon not send it to me after login, or it's mut be from iam (not received from user)?

And why do we need "selling_partner_id"?

**Comment by EvilHagrid** [comment]
Same question

**Comment by saleweaver** [comment, marked as answer]
The only thing that changes in your credentials when requesting data for a client is the refresh token. The customer is authorizing your application to access data on their behalf. 

You can use the sell in partner id as you like, for example to keep track of users. But that's totally up to you.

**Comment by My-CD-ROM** [comment]
Thank you.

**Reply by saleweaver** [reply]
If you're using a credentials file (or env variables), you can override the refresh token when creating a new Client. 

    Reports(refresh_token=<...>)

**Reply by My-CD-ROM** [reply]
and when I will pass client refresh token, he will be able to manage his own seller central account (products, orders etc.)?

**Reply by saleweaver** [reply]
Yes, exactly like that. I have a very simple example of the oauth flow here: https://github.com/saleweaver/sp-api-oauth-demo

Once you get the refresh token, you can request their data.
